src/main/scala/io/iohk/ethereum/blockchain/sync/SyncBlockHeadersRequestHandler.scala

```diff
@@ -0,0 +1,62 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{ActorRef, Props, Scheduler}
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockHeaders, GetBlockHeaders}
+
+class SyncBlockHeadersRequestHandler(
+    peer: ActorRef,
+    val requestMsg: GetBlockHeaders,
+    resolveBranches: Boolean)(implicit scheduler: Scheduler)
+  extends FastSyncRequestHandler[GetBlockHeaders, BlockHeaders](peer) {
+
+  override val responseMsgCode: Int = BlockHeaders.code
+
+  override def handleResponseMsg(blockHeaders: BlockHeaders): Unit = {
+    val headers = if (requestMsg.reverse)
+      blockHeaders.headers.reverse
+     else
+      blockHeaders.headers
+
+    val consistentHeaders = checkHeaders(headers)
+
+    (resolveBranches, consistentHeaders) match {
```

Comment:
I think this would be clearer as a nested `if`. The 2 flags are independent of each other...